tests/models/application.spec.coffee

```diff
@@ -192,6 +192,92 @@ describe 'Application Model:', ->
 					expect(application).to.not.exist
 					done()
 
+	describe '.has()', ->
+
+		it 'should throw if no name', ->
+			expect ->
+				application.has(null, _.noop)
+			.to.throw(errors.ResinMissingParameter)
+
+		it 'should throw if name is not a string', ->
+			expect ->
+				application.has([ 'MyApp' ], _.noop)
+			.to.throw(errors.ResinInvalidParameter)
+
+		it 'should throw if not callback', ->
+			expect ->
+				application.has('MyApp', null)
+			.to.throw(errors.ResinMissingParameter)
+
+		it 'should throw if callback is not a function', ->
+			expect ->
+				application.has('MyApp', [ _.noop ])
+			.to.throw(errors.ResinInvalidParameter)
+
+		describe 'given a logged in user', ->
+
+			beforeEach ->
+				@tokenGetUsernameStub = sinon.stub(token, 'getUsername')
+				@tokenGetUsernameStub.returns('johndoe')
+
+			afterEach ->
+				@tokenGetUsernameStub.restore()
+
+			describe 'given no application', ->
+
+				beforeEach ->
+					@pineGetStub = sinon.stub(pine, 'get')
+					@pineGetStub.returns(Promise.resolve([]))
+
+				afterEach ->
+					@pineGetStub.restore()
+
+				it 'should return false', (done) ->
+					application.has 'MyApp', (error, has) ->
+						expect(error).to.not.exist
+						expect(has).to.be.false
+						done()
+
+			describe 'given an application', ->
+
+				beforeEach ->
+					@application =
```

Comment:
I copy-pasted a chunk from the application.get tests, and this was used in the `it` clause. Safe to remove here.